File: 2ano/2S/AED/prj1/Onn/Onn.py
import time
import elem as arr  # Assuming elem is a module with an array

def verifica(arr):
    # Start timing
    start_time = time.time()
    
    # variables
    length = len(arr)
    minEl = min(arr)
    maxEl = max(arr)
    flag = False
    missing = -1

    # two fors iterate through the loop N² times
    for i in range(length-1):
        for j in range(length-1):
            if(arr[j] == arr[i]+1 and arr[i] < maxEl):
                flag = True
                break
            elif(arr[j] == maxEl and not flag):
                missing = arr[i]+1
    
    # End timing
    end_time = time.time()
    
    # Calculate execution time in seconds
    execution_time = end_time - start_time
    
    print(f"Missing element: {missing}")
    print(f"Execution time: {execution_time:.6f} seconds")
    
    return missing

# Example usage (since I don't have access to elem module)
if __name__ == "__main__":
    # Test with a sample array since elem isn't available
    test_arr = [1, 2, 4, 5, 6]  # Should find 3 as missing
    verifica(test_arr)

File: 2ano/2S/AED/prj1/Onn/Onn.c
#include <stdio.h>
#include <time.h>

// Function to find minimum value in array
int findMin(int arr[], int length) {
    int min = arr[0];
    for(int i = 1; i < length; i++) {
        if(arr[i] < min) min = arr[i];
    }
    return min;
}

// Function to find maximum value in array
int findMax(int arr[], int length) {
    int max = arr[0];
    for(int i = 1; i < length; i++) {
        if(arr[i] > max) max = arr[i];
    }
    return max;
}

int verifica(int arr[], int length) {
    // Start timing
    clock_t start_time = clock();
    
    // Variables
    int minEl = findMin(arr, length);
    int maxEl = findMax(arr, length);
    int flag = 0;    // Using int instead of boolean (0 = false, 1 = true)
    int missing = -1;

    // Two nested loops
    for(int i = 0; i < length-1; i++) {
        for(int j = 0; j < length-1; j++) {
            if(arr[j] == arr[i]+1 && arr[i] < maxEl) {
                flag = 1;
                break;
            }
            else if(arr[j] == maxEl && !flag) {
                missing = arr[i]+1;
            }
        }
        if(flag) break;  // Break outer loop if flag is set
    }
    
    // End timing
    clock_t end_time = clock();
    
    // Calculate execution time in seconds
    double execution_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;
    
    printf("Missing element: %d\n", missing);
    printf("Execution time: %.6f seconds\n", execution_time);
    
    return missing;
}

int main() {
    // Test array (equivalent to Python example)
    int arr[] = {1, 2, 4, 5, 6};
    int length = sizeof(arr) / sizeof(arr[0]);
    
    verifica(arr, length);
    
    return 0;
}
